src/app/app.component.spec.ts

```diff
@@ -1,11 +1,17 @@
+import { HttpClientModule } from '@angular/common/http';
 import { TestBed } from '@angular/core/testing';
-import { RouterTestingModule } from '@angular/router/testing';
+import { BrowserModule } from '@angular/platform-browser';
+import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+import { GatewayService } from './services/gateway.service';
+import { PostService } from './services/post.service';
+import { SubredditService } from './services/subreddit.service';
 
 describe('AppComponent', () => {
     beforeEach(async () => {
         await TestBed.configureTestingModule({
-            imports: [RouterTestingModule],
+            imports: [BrowserModule, AppRoutingModule, HttpClientModule],
```

Comment:
At one point you will have issues with this and you will have to mock the entire HttpClient. As long as you are not doing HTTP calls to server, this is fine. You want to completely isolate unit tests from any other functionality.

Reply:
Alright, if that's okay for now I will do it in new unit test branch